bluemira/geometry/optimisation/problem.py

```diff
@@ -61,7 +61,8 @@ def df_objective(self, geom: _GeomT) -> np.ndarray:
         """
         raise NotImplementedError
 
-    def eq_constraints(self) -> List[GeomConstraintT]:
+    @staticmethod
+    def eq_constraints() -> List[GeomConstraintT]:
         """
```

Comment:
I don't understand; doesn't that mean that all derived classes should follow the same signature? I.e. staticmethod?

Reply:
I take it as the minimum number of inputs to the method is none but any child method can be a normal method (therefore have a self arg) or have any other number of arguments. The base class implements the minimum interface.

Maybe @oliverfunk has some further thoughts?
I'll see if I can find some guidance somewhere.

Reply:
I see it as, marking a method as static means it can (should) only be static for itself and all inheritors of the class.

The shape of the input to the function (i.e. the number and type of the parameters) can change, but the way it's called should stay consistent.
(GeomOptimisationProblem.eq_constraints() vs some_insta.eq_constraints())

A method shouldn't sometimes be static and sometimes be dynamic based on the type imo.

Reply:
Ok, after a bit of followup discussion, I'll take away the staticmethod's for ABCs

Reply:
Done